runtime/drivers/duckdb/olap.go

```diff
@@ -181,6 +186,344 @@ func (c *connection) EstimateSize() (int64, bool) {
+			// not cleaning the source directory for graceful failure of repeate create or replace table statements
+			return err
+		}
+
+		if oldVersionExists {
+			oldDB := dbName(name, oldVersion)
+			// ignore these errors since source has been correctly ingested and attached
+			if err = c.Exec(ensuredCtx, &drivers.Statement{Query: fmt.Sprintf("DETACH %s", safeSQLName(oldDB))}); err != nil {
+				c.logger.Error("detach failed", zap.String("db", oldDB), zap.Error(err))
+			}
+			removeDBFile(filepath.Join(sourceDir, fmt.Sprintf("%s.db", oldVersion)))
+		}
+		return nil
+	})
+}
+
+// DropTable implements drivers.OLAPStore.
+func (c *connection) DropTable(ctx context.Context, name string, view bool) error {
+	if view {
+		return c.Exec(ctx, &drivers.Statement{
+			Query:       fmt.Sprintf("DROP VIEW IF EXISTS %s", safeSQLName(name)),
+			Priority:    100,
+			LongRunning: true,
+		})
+	}
+	if !c.config.TableAsView {
+		return c.Exec(ctx, &drivers.Statement{
+			Query:       fmt.Sprintf("DROP TABLE IF EXISTS %s", safeSQLName(name)),
+			Priority:    100,
+			LongRunning: true,
+		})
+	}
+
+	// ignore query cancellations
+	ctx = context.WithValue(context.Background(), connCtxKey{}, connFromContext(ctx))
+	sourceDir := filepath.Join(filepath.Dir(c.config.DBFilePath), name)
+	version, exist, err := c.tableVersion(name)
+	if err != nil {
+		return err
+	}
+
+	if !exist {
+		return nil
+	}
+	// drop view
+	err = c.Exec(ctx, &drivers.Statement{
+		Query:       fmt.Sprintf("DROP VIEW IF EXISTS %s", safeSQLName(name)),
+		Priority:    100,
+		LongRunning: true,
+	})
+	if err != nil {
+		return err
+	}
+
+	oldDB := dbName(name, version)
+	err = c.Exec(ctx, &drivers.Statement{
+		Query:       fmt.Sprintf("DETACH %s", safeSQLName(oldDB)),
+		Priority:    100,
+		LongRunning: true,
+	})
+	if err != nil {
+		return err
+	}
+
+	// delete source directory
+	return os.RemoveAll(sourceDir)
+}
+
+// InsertTableAsSelect implements drivers.OLAPStore.
+func (c *connection) InsertTableAsSelect(ctx context.Context, name string, byName bool, sql string) error {
+	var insertByNameClause string
+	if byName {
+		insertByNameClause = "BY NAME"
+	} else {
+		insertByNameClause = ""
+	}
+
+	if !c.config.TableAsView {
+		return c.Exec(ctx, &drivers.Statement{
+			Query:       fmt.Sprintf("INSERT INTO %s %s (%s)", safeSQLName(name), insertByNameClause, sql),
+			Priority:    1,
+			LongRunning: true,
+		})
+	}
+	version, exist, err := c.tableVersion(name)
+	if err != nil {
+		return err
+	}
+	if !exist {
+		return fmt.Errorf("table %q does not exist", name)
+	}
+	return c.Exec(ctx, &drivers.Statement{
+		Query:       fmt.Sprintf("INSERT INTO %s.default %s (%s)", safeSQLName(dbName(name, version)), insertByNameClause, sql),
+		Priority:    1,
+		LongRunning: true,
+	})
+}
+
+// RenameTable implements drivers.OLAPStore.
+func (c *connection) RenameTable(ctx context.Context, oldName, newName string, view bool) error {
+	if oldName == newName {
+		// todo check case sensitivity
```

Comment:
Must not check case sensitivity – those actually need to be renamed

Reply:
As per duckdb's implementation we can't rename `test` to `Test` so wondering if we should have the same impl here.

Reply:
That case is already handled in reconcile, try searching for `__rill_tmp_rename_`.

I'm indifferent about whether it's implemented here or in reconcile, it's a pretty rare case (maybe just keep it in reconcile to keep the changes here minimal?).

Reply:
Sure then I guess we need to add case insensitive check.